apps/studio/tests/features/explain-visualizer/ExplainVisualizer.parser.test.ts

```diff
@@ -517,6 +517,95 @@ describe('parseExplainOutput', () => {
       expect(result).toHaveLength(1)
       expect(result[0].details).toContain('Output: id, name, email')
     })
+
+    test('handles invalid cost values gracefully', () => {
+      const input = toQueryPlanRows([
+        'Seq Scan on users  (cost=invalid..notanumber rows=100 width=36)',
+      ])
+
+      const result = parseExplainOutput(input)
+
+      expect(result).toHaveLength(1)
+      expect(result[0].operation).toBe('Seq Scan')
+      // Should not parse invalid cost at all
+      expect(result[0].cost).toBeUndefined()
+    })
+
+    test('handles invalid rows value gracefully', () => {
+      const input = toQueryPlanRows([
+        'Seq Scan on users  (cost=0.00..10.50 rows=notanumber width=36)',
+      ])
+
+      const result = parseExplainOutput(input)
+
+      expect(result).toHaveLength(1)
+      expect(result[0].operation).toBe('Seq Scan')
+      // Should not parse invalid rows at all (regex won't match)
+      expect(result[0].rows).toBeUndefined()
+    })
+
+    test('handles invalid actual time values gracefully', () => {
+      const input = toQueryPlanRows([
+        'Seq Scan on users  (cost=0.00..10.50 rows=100 width=36) (actual time=invalid..notanumber rows=85 loops=1)',
+      ])
+
+      const result = parseExplainOutput(input)
+
+      expect(result).toHaveLength(1)
+      // Should not parse invalid actual time at all
+      expect(result[0].actualTime).toBeUndefined()
+    })
+
+    test('handles invalid rowsRemovedByFilter value gracefully', () => {
+      const input = toQueryPlanRows([
+        'Seq Scan on users  (cost=0.00..10.50 rows=100 width=36)',
+        "  Filter: (status = 'active')",
+        '  Rows Removed by Filter: notanumber',
+      ])
+
+      const result = parseExplainOutput(input)
+      parseNodeDetails(result[0])
+
+      expect(result).toHaveLength(1)
+      // Should not parse invalid value at all (regex won't match)
+      expect(result[0].rowsRemovedByFilter).toBeUndefined()
+    })
+
+    test('ensures all numeric values are finite after parsing', () => {
```

Comment:
Nit: claim is a bit broader than the test ensures, because the metrics are all explicitly defined, so this is more checking, e.g., that `node.cost.start` is `0`, and we know 0 to be finite, than checking that it will never produce an infinite value (that would require a property-based test)

Reply:
I just updated the description to be more granular